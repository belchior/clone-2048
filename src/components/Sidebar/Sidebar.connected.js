import { connect } from 'react-redux';
import { Sidebar } from './Sidebar';

export const mapStateToProps = (state) => {
  return {
    score: state.score,
    bestScore: state.bestScore,
  };
};

export const mapDispatchToProps = () => {
  return {};
};

export default connect(mapStateToProps, mapDispatchToProps)(Sidebar);
