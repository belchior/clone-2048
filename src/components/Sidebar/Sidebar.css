@media all {
  .Sidebar {
    font-family: 'Montserrat Alternates';
  }

  .menu_compact {
    position: absolute;
    top: 0;
    left: 0;
    display: flex;
    width: 100%;
    box-shadow: 0 0 3px rgba(0,0,0,0.4);
    transform: translate3d(0,0,0);
    transition: all 0.3s;
  }
  .menu_compact .btn-menu {
    border: 0;
    padding: 0.5rem;
    width: 3rem;
    height: 3rem;
    background-color: #ffffff;
  }
  .menu_compact .btn-menu img {
    display: block;
  }
  .menu_compact .Score {
    flex: 1;
  }

  .menu_full {
    z-index: 1;
    position: absolute;
    top: 0;
    left: 0;
    display: flex;
    width: 100%;
    height: 100%;
    background-color: rgba(0,0,0,0.4);
    opacity: 0;
    visibility: hidden;
    transition: all 0.3s;
  }
  .menu_full.visible {
    opacity: 1;
    visibility: visible;
  }
  .menu_full.visible .menu_full-container {
    transform: translate3d(0,0,0);
  }

  .menu_full-container {
    overflow-y: auto;
    flex: 0 0 19rem;
    background-color: var(--gray1);
    box-shadow: 0 0 3px rgba(0,0,0,0.4);
    transform: translate3d(-100%,0,0);
    transition: all 0.3s;
  }


  .Sidebar .title {
    display: flex;
    flex-direction: column;
    align-items: center;
    margin: 0;
    padding: 4rem 0 2rem 0;
  }
  .Sidebar .logo {
    margin: 0 0 0.5rem 0;
  }

  .Sidebar .list {
    flex: 1 1 100%;
  }
  .Sidebar .list-item {
    display: flex;
    align-items: center;
    padding: 0.5em;
  }
  .Sidebar .list-item-image {
    display: inline-block;
    margin: 0 0.5em;
    border-radius: 50%;
    width: 2em;
    height: 2em;
  }
  .Sidebar .external-links {
    flex: 0 1;
    margin: auto 0 0 0;
  }
}

@media (min-width: 40rem) {
  .menu_compact {
    transform: translate3d(0,-110%,0);
  }
  .menu_full {
    /* position: static; */
    width: 19rem;
    visibility: visible;
    opacity: 1;
  }
  .menu_full-container {
    transform: translate3d(0,0,0);
  }
}
