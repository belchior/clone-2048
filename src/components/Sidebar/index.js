export * from './Sidebar';
export * from './Sidebar.connected';
export { default } from './Sidebar.connected';
