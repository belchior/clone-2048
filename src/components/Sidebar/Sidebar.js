import React, { Component, } from 'react';
import PropTypes from 'prop-types';

import './Sidebar.css';
import { Score, } from '../Score';
import Controllers from '../Controllers';
import hamburgerMenu from '../../static/images/icon-hamburger-menu.svg';
import logo from '../../static/images/clone-2048.svg';
import iconGithub from '../../static/images/icon-github.svg';

export class Sidebar extends Component {
  state = {
    menuFullVisibility: false,
  };

  handleVisibility = () => {
    /* eslint no-invalid-this: off */
    this.setState(prevState => ({
      menuFullVisibility: !prevState.menuFullVisibility,
    }));
  }

  handleStopPropagation = (event) => {
    event.stopPropagation();
  }

  render() {
    const { bestScore, score, } = this.props;
    const { menuFullVisibility, } = this.state;

    const className = `menu_full ${menuFullVisibility ? 'visible' : ''}`;

    return (
      <aside className="Sidebar">
        <div className="menu_compact">
          <button
            className="btn-menu"
            onClick={this.handleVisibility}
            type="button"
          >
            <img alt="Ã­cone do menu principal" src={hamburgerMenu} />
          </button>
          <Score best={bestScore} current={score} />
        </div>

        <div className={className} onClick={this.handleVisibility}>
          <div className="menu_full-container" onClick={this.handleStopPropagation}>
            <h1 className="title">
              <img alt="Logo clone 2048" className="logo" src={logo} />
              <span>Clone 2048</span>
            </h1>
            <Score best={bestScore} current={score} />
            <Controllers />
            <nav className="list external-links">
              <a className="list-item link-github" href="https://github.com/belchior/clone-2048">
                <img alt="Logotipo do GitHub" className="list-item-image" src={iconGithub} />
                <span className="list-item-text">Calc no GitHub</span>
              </a>
              <a className="list-item" href="https://github.com/belchior">
                <img
                  alt="Fotografia de Belchior Oliveira"
                  className="list-item-image"
                  src="https://avatars0.githubusercontent.com/u/2656585?v=3"
                />
                <span className="list-item-text">Belchior Oliveira</span>
              </a>
            </nav>
          </div>
        </div>

      </aside>
    );
  }
}

Sidebar.propTypes = {
  bestScore: PropTypes.number.isRequired,
  score: PropTypes.number.isRequired,
};
