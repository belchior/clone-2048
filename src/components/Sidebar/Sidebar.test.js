import React from 'react';
import { Sidebar } from './Sidebar';
import ShallowRenderer from 'react-test-renderer/shallow';
import { shallow } from 'enzyme';

it('should render Sidebar without crashing', () => {
  const renderer = new ShallowRenderer();
  const tree = renderer.render(
    <Sidebar bestScore={0} score={0} />
  );

  expect(tree).toMatchSnapshot();
});

it('should make visible menu_full when clicked at hamburger menu', () => {
  const component = shallow(<Sidebar />);
  expect(component.find('.menu_full.visible')).toHaveLength(0);

  component.find('.btn-menu').simulate('click');
  expect(component.find('.menu_full.visible')).toHaveLength(1);
});

it('should hide the menu_full when clicked at root of menu_full (the transparent space)', () => {
  const component = shallow(<Sidebar />);
  expect(component.find('.menu_full.visible')).toHaveLength(0);

  component.find('.btn-menu').simulate('click');
  expect(component.find('.menu_full.visible')).toHaveLength(1);

  component.find('.menu_full').simulate('click');
  expect(component.find('.menu_full.visible')).toHaveLength(0);
});

it('should not hide the menu_full when clicked at body of menu_full', () => {
  const component = shallow(<Sidebar />);
  expect(component.find('.menu_full.visible')).toHaveLength(0);

  component.find('.btn-menu').simulate('click');
  expect(component.find('.menu_full.visible')).toHaveLength(1);

  component.find('.menu_full-container').simulate('click', { stopPropagation: () => undefined });
  expect(component.find('.menu_full.visible')).toHaveLength(1);
});
