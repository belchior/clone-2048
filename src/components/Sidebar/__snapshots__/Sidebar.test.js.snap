// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Sidebar without crashing 1`] = `
<aside
  className="Sidebar"
>
  <div
    className="menu_compact"
  >
    <button
      className="btn-menu"
      onClick={[Function]}
    >
      <img
        alt="Ã­cone do menu principal"
        src="icon-hamburger-menu.svg"
      />
    </button>
    <Unknown
      best={0}
      current={0}
    />
  </div>
  <div
    className="menu_full "
    onClick={[Function]}
  >
    <div
      className="menu_full-container"
      onClick={[Function]}
    >
      <h1
        className="title"
      >
        <img
          alt="Logo clone 2048"
          className="logo"
          src="clone-2048.svg"
        />
        <span>
          Clone 2048
        </span>
      </h1>
      <Unknown
        best={0}
        current={0}
      />
      <Connect(Component) />
      <nav
        className="list external-links"
      >
        <a
          className="list-item link-github"
          href="https://github.com/belchior/clone-2048"
        >
          <img
            alt="Logotipo do GitHub"
            className="list-item-image"
            src="icon-github.svg"
          />
          <span
            className="list-item-text"
          >
            Calc no GitHub
          </span>
        </a>
        <a
          className="list-item"
          href="https://github.com/belchior"
        >
          <img
            alt="Fotografia de Belchior Oliveira"
            className="list-item-image"
            src="https://avatars0.githubusercontent.com/u/2656585?v=3"
          />
          <span
            className="list-item-text"
          >
            Belchior Oliveira
          </span>
        </a>
      </nav>
    </div>
  </div>
</aside>
`;
